src/xrc/xh_bttn.cpp

```diff
@@ -29,9 +29,31 @@ wxButtonXmlHandler::wxButtonXmlHandler()
     XRC_ADD_STYLE(wxBU_TOP);
     XRC_ADD_STYLE(wxBU_BOTTOM);
     XRC_ADD_STYLE(wxBU_EXACTFIT);
+    XRC_ADD_STYLE(wxBU_NOTEXT);
     AddWindowStyles();
 }
 
+// Function calls the given setter with the contents of the node with the given
+// name, if present.
+//
+// If alternative parameter name is specified, it is used too.
+void
+wxButtonXmlHandler::SetBitmapIfSpecified(wxButton* button,
```

Comment:
Could we avoid duplicating this function between here and `wxBitmapButtonXmlHandler`? It's not much code, but it still would be better to have it in a single place and it looks like it ought not be too difficult to just make it take `wxAnyButton`.

Reply:
Actually, if we don't support the older alternative names for wxButton, then the function could be eliminated entirely -- which would also eliminate the call to GetBitmap() for the older parameter name (the newer parameters call GetBitmapBundle()). So instead, the code would be:

```c++
    auto node = GetParamNode("pressed");
    if (node)
        button->SetBitmapPressed(GetBitmapBundle(node));
    node = GetParamNode("focus");
    if (node)
        button->SetBitmapFocus(GetBitmapBundle(node));
    node = GetParamNode("disabled");
    if (node)
        button->SetBitmapDisabled(GetBitmapBundle(node));
    node = GetParamNode("current");
    if (node)
        button->SetBitmapCurrent(GetBitmapBundle(node));

```

The other approach would be to make the function a static member of the wxBitmapButtonXmlHandler class, and call it directly. The one downside is that those alternative names (`selected` and `hover`) don't support bundles -- so calling the older wxBitmapButtonXmlHandler method could cause problems for people expecting it to support bundles.

Reply:
Just saw the comment about not using `auto` -- will change accordingly.

Reply:
I'm fine with this, i.e. I think that while the rationale about supporting the deprecated synonyms for the existing XRC using `wxBitmapButton` changed to use `wxButton` makes sense, we don't really _have_ to do it like this and, thinking more about it, it might actually be better not to do it to avoid encouraging keeping using the old names.

So, to summarize, yes, I'd be perfectly fine with just this, thank you.